chapter/src/main/java/com/nk/DetailsResorce.java

```diff
@@ -1,63 +1,80 @@
     DetailsRepository repo =new DetailsRepository();
+
     @POST
     @Path("signup")
     @Consumes({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
     public Response userSignup (UserDetails user) {
-        System.out.println("getting called");
-        String output = "Created customer <a href=\"customers/" +
-                user.getUsername() + "\">" + user.getUsername()
-                + "</a>";
-        String lastVisit = DateFormat.getDateTimeInstance(
-                DateFormat.SHORT, DateFormat.LONG).format(new Date());
-        if(repo.signup(user)){
+        if(repo.signup(user))
+        {
         return Response.created(URI.create("/users/"
                 + user.getUsername()))
-                .entity(output)
-                .cookie(new NewCookie("last-visit", lastVisit))
+                .entity("You have successfully created an account , please login to continue ..").status(201).type("application/json")
                 .build();
-    }return  Response.serverError().build();
+        }
+    return  Response.serverError().build();
     }
 
     @POST
     @Path("login")
     @Consumes({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
-//    @Consumes({MediaType.APPLICATION_FORM_URLENCODED})
     public Response userLogin (UserDetails user) {
         System.out.println("getting called");
-        ;
-        if(repo.login(user))
+      int flag=repo.login(user);
+        if(flag>0)
         {
-            String output = " customer is  <a href=\"customers/" +
-                user.getEmail() + "\">" + user.getEmail()
-                + "</a>";
-        String lastVisit = DateFormat.getDateTimeInstance(
-                DateFormat.SHORT, DateFormat.LONG).format(new Date());
-
         return Response.created(URI.create("/users/"
                 + user.getUsername()))
-                .entity(output)
-                .cookie(new NewCookie("pulseId", user.getEmail()))
+                .entity("You have successfully logged in")
+                .cookie(new NewCookie("pulseId", String.valueOf(flag)))
                 .build();
-    } return Response.status(Response.Status.NOT_FOUND)
-                .entity("NoT FOUND, PLease signup")
-                .type("text/plain").build();
+        }
+        else{
+             if (flag== -1)
+                 return Response.status(Response.Status.UNAUTHORIZED)
+                         .entity("Email or password is incorrect")
+                         .type("application/json").build();
+        }
+    return Response.status(Response.Status.NOT_FOUND)
+                .entity("email does not exists, please signup")
+                .type("application/json").build();
     }
 
+    @POST
+    @Path("adduser")
+    @Consumes({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+     public Response createUser (UserDetails userDetails, @CookieParam("pulseId") int adminId)
+    {
+        if (adminId>0)
+        {
+        repo.addUser(userDetails.getEmail(),adminId);
```

Comment:
Don't take it from cookie. Have a cookie table to store cookie id and admin's ID and set the primary key of the entry in cookie. Do a reverse lookup when u get add user request